regexp.go

```diff
@@ -235,17 +235,14 @@ func (re *Regexp) getRunesAndStart(s string, startAt int) ([]rune, int) {
 		ret[i] = r
 		i++
 	}
+	if startAt == len(s) {
```

Comment:
Interesting point. Currently `startAt` represents a byte offset in a UTF-8 string, but all the result indices are rune offsets... Rewriting it like you suggested would change the semantics and may break other people's code. 

Also, if the caller knows the rune offset it's highly likely that they have the string converted to `[]rune` anyway in which case they would rather use `FindRunesMatchStartingAt` which I ended up using. And this makes this method kind of pointless...

I don't know what to do really 😄  I'm happy to just leave things unchanged as I said I ended up not using this method, just thought the bug was worth fixing...

BTW, the same thing about result indices applies to all *String* methods... Ideally those indices need to be converted back to byte offsets.

Reply:
Ha, been a while since I looked at this code.  Of course you're right.  The code I eliminated in my example was to map the incoming byte index to the outgoing rune index. 

I think the usage of the indices into the rune slice throughout the API is a byproduct of the porting process from .NET.  In C# they can assume string indices are the same as runes...so I decided to punt on that and map it at the start instead of rooting out any deep assumptions.

Reply:
What should I do about this PR then?